### ISSUE TITLE
Consider using mybinder

### ISSUE THREAD

**Issue by fedorov** [Radiomics/pyradiomics#275]
See https://beta.mybinder.org/

**Comment by fedorov** [comment]
Currently building of the binder is failing due to #276

**Comment by jcfr** [comment]
If I recall, binder is not maintained anymore.

Cc: @thewtex

**Comment by pieper** [comment]
It's not currently working for me (constant spinning logo and server error
in the browser console).

**Comment by jcfr** [comment]
If you check the status, it is down: http://mybinder.org/status

@thewtex should be able to confirm, but if i recall they run out of funding. The model was not sustainable ( to be confirmed)

**Comment by fedorov** [comment]
See the URL I included in the initial post ;-)

**Comment by fedorov** [comment]
cc: @yuvipanda

**Comment by fedorov** [comment]
Working example: https://github.com/minrk/ligo-binder and direct binder url https://beta.mybinder.org/v2/gh/minrk/ligo-binder/master?filepath=index.ipynb

**Comment by jcfr** [comment]
Indeed, there are mybinder.org and beta.mybinder.org

I just wonder what is backing up beta.mybinder.org.

**Comment by pieper** [comment]
@fedorov your original post didn't link to a working example!

**Comment by fedorov** [comment]
indeed. What I meant to imply in an awkward manner is that I was not referring to http://mybinder.org. https://beta.mybinder.org/ can presumably be used for any repo that has a python notebook, so I was not sure what exactly was not working and timed out from https://beta.mybinder.org/.

**Comment by yuvipanda** [comment]
Heya! Me and @choldgraf and others from the [Jupyter Project](https://jupyter.org) have started assisting mybinder, and rewrote it to run on top of [JupyterHub](http://github.com/jupyterhub/jupyterhub/). beta.mybinder.org is the newer version, and should be a lot more stable (and we're committed to supporting it, and have structural support to doing so). The long term hope is that we make it really easy for institutions, publications and organizations to set up their own binder install, so people can use the same compatible repositories on any of them.

Am happy to help answer any other questions!

**Comment by pieper** [comment]
Thanks for the info @yuvipanda!  This will be very helpful.

**Comment by thewtex** [comment]
@yuvipanda That is fantastic to hear! For organizations that do not have the resources to set up their own JupyterHub instance, how many resources are available and how long will they be available for?

**Comment by choldgraf** [comment]
+1 to what Yuvi said!

To @thewtex's question, Binder is a space that's evolving relatively rapidly right now so it's hard to say exactly what resources etc are available. That said, there's definitely a lot of interest in making the platform as easy as possible to deploy on other servers etc, and also a lot of interest in keeping the public `mybinder.org` service running as a public good. Feel free to open an issue over on the binderhub repo if there's a specific configuration setup you're concerned about (https://github.com/jupyterhub/binderhub)

**Comment by JoostJM** [comment]
Using the branch from #285, I was able to build and launch using MyBinder.
This required some changes to the dockerfile, as 3 notebooks are run as part of the build process, which assume a certain structure of the files and folders.

**Comment by JoostJM** [comment]
This is the [link](https://beta.mybinder.org/v2/gh/JoostJM/pyradiomics/fix-urllib-python3-compatibility) to the image.

**Comment by choldgraf** [comment]
just a quick thought - Binder now supports new features of repo2docker, including the ability to run shell scripts prior to generating the Docker image. So if you'd like to handle pre-build stuff with scripts instead of a Dockerfile, that should be possible now!

**Comment by JoostJM** [comment]
[link](https://beta.mybinder.org/v2/gh/radiomics/pyradiomics/master) to the image build upon radiomics/pyradiomics/master

**Comment by fedorov** [comment]
FYI another free cloud-based notebook hosting platform: https://notebooks.azure.com

Disadvantages:
* need to have a login and an account to run a notebook (since it first has to be cloned)
* no support for Dockerfile for configuration

Advantages:
* new features seem to be added regularly, seems to be actively supported: https://notebooks.azure.com/new
* may be more robust than beta.mybinder, since it has MS/Azure back-end

**Comment by choldgraf** [comment]
FWIW, Binder is actively supported as well ;-)  (this wasn't true 9 months ago but it's a jupyter project now)

Also I'm curious what you mean by "more robust"?

(full disclosure: I'm on the jupyter team and working on the binder project :-) )

**Comment by fedorov** [comment]
@choldgraf sorry, I didn't mean to offend you! I have a lot of respect to Binder, and it has some significant advantages!

What I meant by "actively supported" is that is has Microsoft and Azure backing it, so there is a clear entity with deep pockets paying for the cloud cycles. I also liked that there is a timeline with updates which seems to be updated quite often. With Binder, documentation seems to be quite dispersed, it is not clear how to do things beyond the example binders, and what is going on with the development. It was also confusing to me as a beginner to find my way around (I have not tried very hard, I admit) the soup of terms "binder", "beta binder", "jupyter hub", "docker2repo" - it is quite hard to understand how to navigate this and what is going on. BUT (and that's a HUGE "but"!!!) once I get something that "just works" - I am pretty happy - as long as it works - and I don't care that much how and why it works.

About robustness - I was referring to the discussion we had on gitter with @minrk. Yesterday I had a mysterious failure to launch, with no error messages, which you guys seemed to have traced down to github throttling due to lack of OAuth integration. @minrk seemed to suggest that with the Azure Notebooks there would be no problems loading (and that's understandable given how much resources MS has).

**Comment by choldgraf** [comment]
absolutely not offended :-) just trying to figure out where binder isn't meeting expectations so that we can figure out if/how we should try to improve things!

I totally agree about all the issues you just noted - some of them are just because the tech behind Binder is definitely still in a beta state...this stuff didn't exist in its current form > 4 months ago :-)  if there's something in particular that's confusing etc in the documentation, don't hesitate to open an issue in the `binder` or `binderhub` repos. One of our next focuses is to build up the documentation.

**Comment by fedorov** [comment]
@choldgraf I think a good start would be to add a link to the documentation / "getting started" / examples to the front page: https://beta.mybinder.org/. It is quite confusing that the front page does not have any pointer to the documentation!

**Comment by choldgraf** [comment]
that's a good point! I agree that should be posted there...

FWIW, here are the docs we'll iterate on for the beta deployment: http://mybinder.readthedocs.io/en/latest/
